src/WebViewShared.tsx

```diff
@@ -34,6 +34,14 @@ const matchWithRegexList = (
   return compiledRegexList.some(x => x.test(value));
 };
 
+const matchWithPrefixStringList = (
+  prefixes: readonly string[],
+  value: string,
+) => {
+  if (typeof value !== 'string') throw new Error(`value was not a string`)
+  return prefixes.some(x => value.startsWith(x));
```

Comment:
`String.prototype.startsWith` returns `true` if the given characters are found at the beginning of the string, including when `searchString` is an **empty string**; also, all `searchString` values that are not regexes are **coerced to strings** ([MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith)).

So
```js
prefixes.some(x => value.startsWith(x))
```
can be bypassed with
```js
prefixes = ['']
```
And
```js
prefixes.some(x => x.length && value.startsWith(x))
```
can be bypassed with
```js
prefixes = [['']]
```

---
Hardened implementation:
```suggestion
  return prefixes.some(x => String(x).length && String.prototype.startsWith.call(value, x));
```

Reply:
The proposed impl. still suffers from weakness. The point is that without additional checks, it's still possible to get all deep links approved (`['a', 'b', 'c', 'd', 'e', 'f', 'u']`). @ChALkeR, thoughts on how to harden it further?